src/api.js

```diff
@@ -0,0 +1,110 @@
+'use strict';
+const assert = require('assert');
+const brorand = require('brorand');
+const hashjs = require('hash.js');
+const elliptic = require('elliptic');
+const Ed25519 = elliptic.eddsa('ed25519');
+const Secp256k1 = elliptic.ec('secp256k1');
+const addressCodec = require('ripple-address-codec');
+const derivePrivateKey = require('./secp256k1').derivePrivateKey;
+const accountPublicFromPublicGenerator = require('./secp256k1')
+  .accountPublicFromPublicGenerator;
+const utils = require('./utils');
+const hexToBytes = utils.hexToBytes;
+const bytesToHex = utils.bytesToHex;
+
+function generateSeed(options = {}) {
+  assert(!options.entropy || options.entropy.length >= 16, 'entropy too short');
+  const entropy = options.entropy ? options.entropy.slice(0, 16) : brorand(16);
+  return addressCodec.encodeSeed(entropy, options.algorithm || 'secp256k1');
```

Comment:
Where do you plan to support passphrase -> seed generation, if at all? People only really should use it for testing, so I could buy the argument that you don't want it in this new api, however it would be convenient.